### PR TITLE
Enable Template helpers

### PR DESCRIPTION
**Pull request by thetechnick**
### Summary
Allow users to define template blocks to reuse template blocks.

Also stops the test-fixture generator to create empty files, which becomes annoying for helper files just containing define blocks.

### Change Type
New Feature

### Check List Before Merging

- [x] This PR passes all pre-commit hook validations.
- [x] This PR is fully tested and regression tests are included.
- [ ] Relevant documentation has been updated.

### CODE REVIEW

File: docs/api-reference.md [hunk 1]
```diff
@@ -29,6 +29,7 @@ apiVersion: package-operator.run/v1alpha1
 kind: ClusterObjectDeployment
 metadata:
   name: example
+  namespace: default
```
Comment (eqrx) [inline docs/api-reference.md:32]: this file is not part of the PR

**Review by eqrx — CHANGES_REQUESTED** [review]

File: docs/api-reference.md (hunk 1 above)
Comment (erdii) [reply docs/api-reference.md:32]: fix: rebase onto main and redo the commit, k8s-docgen has been fixed

**Review by eqrx — APPROVED** [review]
